documentation/book/oauth/proc-oauth-server-config.adoc

```diff
@@ -0,0 +1,61 @@
+// Module included in the following module:
+//
+// con-oauth-config.adoc
+
+[id='proc-oauth-server-config-{context}']
+= Configuring an {oauth} authorization server
+
+This procedure describes how to configure an authorization server for integration with {ProductName}.
+
+Depending on the chosen xref:con-oauth-authentication-client-options-{context}[authentication flow], you can use the authorization server to:
+
+* Configure authentication for Kafka brokers
+* Configure and authorize clients
+* Configure users and roles
+* Obtain access and refresh tokens
+
+User credentials are managed by an authorization server using a _realm_, which describes security policies.
+The _realm_ is the authentication provider for registered clients.
+
+To use {oauth} with {ProductName}, you need a deployment of an authorization server to be able to create and manage authentication realms.
+
+NOTE: If you already have an authorization server deployed, you can skip the deployment step and use your current deployment.
+
+.Prerequisites
+
+* {ProductName} and Kafka is running
+
+For an authorization server deployment:
+
+* Check the supported configuration for the authorization server
+* Installation requires a user with a cluster-admin role, such as system:admin
+
+.Procedure
+
+. Deploy the authorization server to your cluster.
+. Access the CLI or admin console for the authorization server to create the {oauth} policies for {ProductName}.
++
+Consult the product documentation for the authorization server for information on creating policies.
+
+. Create and enable a _realm_.
+
+. Adjust the session and token timeouts for the _realm_, if required.
+
+. Create a _client_ called `kafka-broker`.
++
+--
+* Set the client type to `confidential`
+* Enable client OpenID Connect (OIDC) authorization
+* Disable client OpenID Connect (OIDC) authentication
+--
+
+. Take a note of the _client_ secret for using in your {ProductName} Kafka cluster configuration.
+
+. Repeat the _client_ creation steps for any application client that will connect to your Kafka brokers.
++
+Create a definition for each new _client_.
++
+You will use the names as client IDs in your configuration.
```

Comment:
These are Keycloak specific instructions. If we want to give Keycloak specific instructions, then we need a whole chapter with very specific instructions, referring to specific Admin Console elements.
I suggest we remove all these lines.

Reply:
@mstruk Thanks. But now we have no procedure! I guess we need some generic outline at least? Deploy an authorization server, configure access from client to broker according to the chosen authentication flow, make sure that you have this or that configured?
@jakub - do we prefer to have no steps and keep it very generic? Or maybe we show by example, with keycloak steps?

Reply:
I think that needs wider discussion also with @tombentley and @ppatierno as maintainers. In general I do not have problem with Keycloak specific instructions (Keycloak and not RH-SSO here). But will let Tom and Paolo comment as well. We can alternatively remove it here and cover it with blogposts for Keycloak, Hydra, and anything else which comes later to our mind.

Reply:
We can stick with Marko's suggestion to just have:

Preparing authorization server to work with Strimzi involves the following steps:

* Configuring 'kafka-broker' client
* Configuring clients for each Kafka client component of your application

And then link to blogposts for the specifics.

Reply:
Here's a more generic procedure. Maybe this is enough for Tech Preview? Then we can decide whether to introduce keycloak or link to blog posts with the specifics as they come online. 

![image](https://user-images.githubusercontent.com/47596553/66374978-21a6d200-e9a4-11e9-88bc-40f0f6b9fc26.png)

Reply:
That sounds reasonable to me. 👍

Reply:
We need to strike the right balance between:

1. Having enough documentation that users are able to set it up correctly with the authentication server of their choice.
2. Having documentation which we can maintain as a project.
3. The Strimzi project not looking like it's promoting "Red Hat" projects at the expense of other popular alternatives (i.e. just providing detailed doc for Keycloak is not an option imho).

Admin UIs sometimes get refreshed or changed. It's quite likely we wouldn't know about such changes in other products until someone opened a bug about it, and then it can be quite a lot of work to re-work the documentation. So I think there is a real cost to the both our users and contributors in trying to document multiple auth servers in detail.

For this reason I think the right balance would be to:

* Have generic instructions in the doc. I think this is acceptable because presumably each auth server comes with its own doc which covers setting it up with clients, so users should be able to put the two bits of doc together themselves.
* Blog about how to set it up with the more popular auth servers. People don't expect blog posts to be maintained in the same way they do about documentation, so I think this doesn't entail a maintenance burden. Crucially, we would need to blog about more than just Keycloak to not fall foul of point 3.
* If Red Hat want better documentation for how to do it with Keycloak/RH SSO that can be done outside of the Strimzi project.

Reply:
Thanks all. Okay. I have made it clear in the intro that the steps are generic and the actual steps dependent on the authorization server. 

@mstruk - Please check the updated file with the understanding we are staying generic.